plugins/builder-pack/builder-build

```diff
@@ -31,11 +31,12 @@ trigger-builder-pack-builder-build() {
   local TMP_FILE=$(mktemp "/tmp/dokku-${DOKKU_PID}-${FUNCNAME[0]}.XXXXXX")
   trap "rm -rf '$TMP_FILE' >/dev/null" RETURN INT TERM
 
-  config_export app "$APP" --format envfile --merged >"$TMP_FILE"
+  config_export app "$APP" --format pretty --merged >"$TMP_FILE"
```

Comment:
Could you change this to an eval like [here](https://github.com/dokku/dokku/blob/master/plugins/builder-herokuish/builder-build#L32) (probably with --merged of course to get globals, which makes me think we're missing merged for herokuish...) and then fetch the extra build args [like so](https://github.com/dokku/dokku/blob/0b3454412542c510c482eb6bbb42dc3e0c1ab579/plugins/config/docker-args-deploy#L15)? The `docker-args-keys` format is the same between `pack build` and `docker-build`. This should remove the need for the extra sed call and also properly handle newlines imo.

Well, at least I'm hoping `pack build` supports `--env KEY` the same way `docker build` does.

Reply:
Makes total sense to keep the approaches consistent. I've had a shot at it, and annoyingly, there's _just enough_ discrepancy between the docker args format and the pack args format that I don't think there's going to be a way to avoid string manipulation of some sort somewhere.

Using
```sh
ENV_ARGS="$(config_export app "$APP" --format docker-args-keys --merged)"
eval "$(config_export app "$APP" --merged)"
pack build "$IMAGE" --builder "$DOKKU_CNB_BUILDER" --path "$SOURCECODE_WORK_DIR" \
--default-process web "$ENV_ARGS"
```

The result is:
```
# assuming FOO=foo and BAR=bar in ENV
pack build my-image:version --builder my-builder --path /src --default-process web --env=FOO --env=BAR
```

Called like this, pack will build, but it ignores those `env` declarations because it expects a space, not an equals sign for its arguments :disappointed: 

I reworked my solution based on your suggestions, latest commit still does some string manipulation :slightly_frowning_face: but I released the dependency on `sed`, at least.

Reply:
I refactored the MR a bit to use my idea with a new export format for pack.